deepspeed/ops/fp_quantizer/quantize.py

```diff
@@ -0,0 +1,79 @@
+# Copyright (c) Microsoft Corporation.
+# SPDX-License-Identifier: Apache-2.0
+
+# DeepSpeed Team
+
+import torch
+
+from deepspeed.ops.op_builder import FPQuantizerBuilder
+
+fp_quant_module = None
+
+
+class FP_Quantize:
+
+    def __init__(self, group_size=512) -> None:
+        global fp_quant_module
+        if fp_quant_module is None:
+            fp_quant_module = FPQuantizerBuilder().load()
+
+        self.group_size = group_size
+        self.orig_dtype = None
+
+    def quantize(self,
+                 input,
+                 q_bits=8,
+                 q_mantisa_bits=3,
```

Comment:
NIT: `q_mantisa_bits` seems only used for `q_bits == 8`. This parameter can be moved and directly assign its value in L37? Or moving the predefined `q_mantisa_bits ` values in L38-43.

Reply:
that's a good point @JamesTheZ, the reason we are receiving as the parameter for FP8 is that we can switch between E5M2 and E4M3, but for the rest we hard-code the bit-ratios the kernel supports.